src/combinatorics/simple_graph/ends/defs.lean

```diff
@@ -198,6 +205,43 @@ end
 
 end component_compl
 
+/-
+For a locally finite preconnected graph, the number of components outside of any finite set
+is finite.
+-/
+lemma component_compl_finite [locally_finite G] (Gpc : preconnected G) (K : finset V) :
+  finite (G.component_compl K) :=
```

Comment:
is this `_root_.finite` or `set.finite`?

Reply:
`set.finite` is protected, so it's the root one.

Reply:
In that case, do you think this should be stated with `[fact (preconnected G)]` so that it can be an instance?

Reply:
I can't tell without knowing the applications of this PR.

Reply:
It would be very useful to have `finite` be an instance, since this would allow calling lemmas/defs from [this PR](https://github.com/leanprover-community/mathlib/pull/18433) more transparently, no?

Reply:
[connected_space](https://leanprover-community.github.io/mathlib_docs/topology/connected.html#connected_space), [quiver.rooted_connected](https://leanprover-community.github.io/mathlib_docs/combinatorics/quiver/arborescence.html#quiver.rooted_connected), [category_theory.is_(pre)connected](https://leanprover-community.github.io/mathlib_docs/category_theory/is_connected.html#category_theory.is_connected) are already classes, but [quiver.is_preconnected](https://leanprover-community.github.io/mathlib_docs/combinatorics/quiver/symmetric.html#quiver.is_preconnected) and [simple_graph.(pre)connected](https://leanprover-community.github.io/mathlib_docs/combinatorics/simple_graph/connectivity.html#simple_graph.connected) are not. Should we make them classes?

Reply:
I evoked the idea [in this thread](https://leanprover.zulipchat.com/#narrow/stream/252551-graph-theory/topic/connectivity/near/316452420) already: I (perhaps naively) think it's a sensible idea.

Reply:
My take is that it's somewhat painful in Lean 3 because of frozen instances. But in Lean 4 there's no problem and hopefully we won't do much more with these before mathlib4, so sure why not.